<launch>
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
      <arg name="world_name" value="$(find urdf7)/world/arena3.world"/>

  </include>
  <node
    name="tf_footprint_base"
    pkg="tf"
    type="static_transform_publisher"
    args="0 0 0 0 0 20 base_link base_footprint 40" />
  <node
    name="spawn_model"
    pkg="gazebo_ros"
    type="spawn_model"
    args="-file $(find urdf7)/urdf/urdf7.urdf -urdf -model urdf7"
    output="screen" />
  <node
    name="fake_joint_calibration"
    pkg="rostopic"
    type="rostopic"
    args="pub /calibrated std_msgs/Bool true" />
  
  <rosparam file="$(find urdf7)/config/joint_controllers.yaml" command="load"/>
  
  <param name="robot_description" command="cat $(find urdf7)/urdf/urdf7.urdf"/>
  
    <!-- load the controllers -->
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
    output="screen" ns="/urdf7" args="joint_state_controller arm_controller "/>
  <node name="robot_joint_teleop" pkg="urdf7" type="robot_joint_teleop.py" output="screen"/>

    

  <!-- convert joint states to TF transforms for rviz, etc -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"
    respawn="false" output="screen">
    <remap from="/joint_states" to="/urdf7/joint_states" />
  </node>
  
  
</launch>
